### PR TITLE
Adding support for evaluating video classifications

### PR DESCRIPTION
**Pull request by brimoor**
This PR adds support for running classification evaluation on frame-level labels.

I also added a `select_fields` flag to `SampleCollection.load_evaluation_view()` that will automatically select only the fields involved in the evaluation, excluding any other evaluation fields (even embedded fields on the same GT/pred fields that were populated by other evaluation runs). The utility here is that it enables you to easily load up a view that only contains the results of one evaluation, in case you have run multiple evaluations on the same dataset. 

In order to implement the frame-level evaluation cleanly, I invested some time in upgrading the `SampleCollection.set_values()` method so that it now supports setting batches of frame-level fields and batches of sample- or frame-level array fields. So, basically `set_values()` is now a proper inverse of the `values()` aggregation.

An `evaluation_tests.py` module is added that runs (intensive) integration tests of all supported evaluation methods.

Example usage of frame-level evaluation:

```py
import random

import numpy as np

import fiftyone as fo
import fiftyone.zoo as foz
from fiftyone import ViewField as F

ANIMALS = [
    "bear", "bird", "cat", "cow", "dog", "elephant", "giraffe",
    "horse", "sheep", "zebra"
]
NUM_CLASSES = len(ANIMALS)

dataset = foz.load_zoo_dataset("quickstart-video").clone()

with fo.ProgressBar() as pb:
    for sample in pb(dataset):
        for frame in sample.frames.values():
            gt_idx = random.randint(0, NUM_CLASSES - 1)

            logits = np.random.rand(NUM_CLASSES)
            logits = np.log(logits / np.sum(logits))
            if random.random() < 0.8:
                logits[gt_idx] = 0.0

            gt_label = ANIMALS[gt_idx]
            pred_label = ANIMALS[np.argmax(logits)]

            frame["gt_animal"] = fo.Classification(label=gt_label)
            frame["pred_animal"] = fo.Classification(
                label=pred_label, logits=logits
            )

        sample.save()

#
# Multiclass classification
#

EVAL_KEY = "eval_class"

results = dataset.evaluate_classifications(
    "frames.pred_animal", gt_field="frames.gt_animal", eval_key=EVAL_KEY
)

results.print_report()
results.plot_confusion_matrix()

print(dataset.bounds(EVAL_KEY))
print(dataset.count_values("frames." + EVAL_KEY))

print(dataset.list_evaluations())
print(dataset.get_evaluation_info(EVAL_KEY))

#
# Top-k classification
#

TOP_K_EVAL_KEY = "eval_top_k"

results = dataset.evaluate_classifications(
    "frames.pred_animal",
    gt_field="frames.gt_animal",
    eval_key=TOP_K_EVAL_KEY,
    method="top-k",
    classes=ANIMALS,
    k=2,
)

results.print_report()
results.plot_confusion_matrix()

print(dataset.bounds(TOP_K_EVAL_KEY))
print(dataset.count_values("frames." + TOP_K_EVAL_KEY))

print(dataset.list_evaluations())
print(dataset.get_evaluation_info(TOP_K_EVAL_KEY))

#
# Binary classification
#

BINARY_EVAL_KEY = "eval_binary"

binarize = (F("label") == "cat").if_else(F("label"), "other")

binary_view = (
    dataset
    .set_field("frames.gt_animal.label", binarize)
    .set_field("frames.pred_animal.label", binarize)
)

print(binary_view.count_values("frames.gt_animal.label"))
print(binary_view.count_values("frames.pred_animal.label"))

binary_results = binary_view.evaluate_classifications(
    "frames.pred_animal",
    gt_field="frames.gt_animal",
    eval_key=BINARY_EVAL_KEY,
    method="binary",
    classes=["other", "cat"],
)

binary_results.print_report()
binary_results.plot_confusion_matrix()

print(dataset.bounds(BINARY_EVAL_KEY))
print(dataset.count_values("frames." + BINARY_EVAL_KEY))

print(dataset.list_evaluations())
print(dataset.get_evaluation_info(BINARY_EVAL_KEY))

#
# Cleanup
#

print(dataset.list_evaluations())
dataset.delete_evaluations()
print(dataset.list_evaluations())
```

Example usage of `load_evaluation_view()`:

```py
import fiftyone as fo
import fiftyone.zoo as foz
from fiftyone import ViewField as F

EVAL_KEY1 = "eval1"
EVAL_KEY2 = "eval2"

dataset = foz.load_zoo_dataset("quickstart").clone()
results1 = dataset.evaluate_detections("predictions", eval_key=EVAL_KEY1)

view = dataset.take(100)
results2 = view.evaluate_detections("predictions", eval_key=EVAL_KEY2, iou=0.5)

eval_view1 = dataset.load_evaluation_view(EVAL_KEY1, select_fields=True)
eval_view2 = dataset.load_evaluation_view(EVAL_KEY2, select_fields=True)

session = fo.launch_app(dataset)

# View *only* results from first evaluation
session.view = (
    eval_view1
    .sort_by(EVAL_KEY1 + "_fp", reverse=True)
    .filter_labels("predictions", F(EVAL_KEY1) == "fp")
)

# View *only* results from second evaluation
session.view = (
    eval_view2
    .sort_by(EVAL_KEY2 + "_fp", reverse=True)
    .filter_labels("predictions", F(EVAL_KEY2) == "fp")
)
```

### CODE REVIEW

**Review by benjaminpkane — APPROVED** [review]
LGTM :rocket:

**Review by benjaminpkane — APPROVED** [review]
LGTM :rocket:
